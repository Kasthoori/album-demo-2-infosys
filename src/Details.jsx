/* eslint-disable no-undef */
import React, { useState, useEffect } from "react";
import { useLocation, useHistory } from "react-router-dom";
import Header from './components/header/Header';
import { Table } from 'react-bootstrap';
import ListItemImg from './components/ListItemImg/ListItemImg';
import Image from 'react-bootstrap/Image'



const Details = () => {

  const location = useLocation();
  const history = useHistory();

  let status = "";
  let albumtitle = "";
  let userId = "";
  let aId = "";
  let tracks = [];

  const [albumpic, setAlbumPic] = useState("");
  const [albumId, setAlbumId] = useState("");
  const [id, setId] = useState("");
  const [title, setTitle] = useState("");

  try {
    status = location.state.status;
    albumtitle = location.state.title;
    userId = location.state.userId;
    tracks = location.state.tracks;
    aId = location.state.id;
  } catch (err) {
    status = "false";
    tracks = [];
  }

  useEffect(() => {
    if (status !== "true") {
      history.push("/");
    }

    if (tracks.length > 0) {
      setAlbumPic(tracks[0].url);
      setAlbumId(tracks[0].albumId);
      setTitle(tracks[0].title);
      setId(tracks[0].id);
      console.log("YES");
    }
  }, [history, status, tracks]);

  const listItemImgOnClick = (track) => {
    setAlbumPic(track.url);
    setAlbumId(track.albumId);
    setId(track.id);
    setTitle(track.title);
  };

    return (

        <div>
            <span data-testid="headerdetails"><Header /></span>
            <br />
            <br />
            <div className="container">
                <div className="row">
                    <div className="col-md-4" data-testid="detailtable">
                    
                        <Table className="table table-striped table-bordered table-sm" cellspacing="0">
                            <thead>
                            <tr>
                                <th>Details</th>
                            </tr>
                            </thead>
                            <tbody>
                              
                               <tr>
                                    <span data-testid="thumbnailimage">
                                      {tracks.map((track) => (
                                           <ListItemImg data-testid="listimglink" track={track} onclick={listItemImgOnClick} />
                                         ))}
                                      </span>
                                </tr>
                            </tbody>
                        </Table>
                    </div>
                    </div>
                    <div className="col-md-8">
                        <div>
                            <h2>Album ID: {albumId}</h2>
                                <Image data-testid="albumpicture" src={albumpic} alt="" fluid />
                                <span id="idtext">Image ID: {id}</span>
                                <br />
                                <span>Title: {title}</span>
                        </div>
                    </div>
                </div>
            
        </div>

    );

}


export default Details;