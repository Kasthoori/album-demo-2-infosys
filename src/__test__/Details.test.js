/* eslint-disable no-unused-labels */
/* eslint-disable no-labels */
import React from 'react';
import { shallow } from 'enzyme';
import Details from '../../src/Details';
import toJson from 'enzyme-to-json';


jest.mock("react-router-dom", () => ({
    useLocation: jest.fn().mockReturnValue({
      pathname: "",
      search: "",
      hash: "",
      state: null,
      key: "",

    }),
    useHistory: () => {
      push: jest.fn()
    },

    

}));

describe('Details Component', () => {

    //test case 01. number of headers
    test('render header of details', () => {
        const wrapper = shallow(<Details />);
        expect(wrapper.find('[data-testid="headerdetails"]').length).toBe(1);
    });

    //test case 02. number of divs in component
    test('number of div tags', () => {
        const wrapper = shallow(<Details />);
        expect(wrapper.find('div').length).toBe(6);
    });

    //test case 03. render detail table
    test('render details table',() => {
        const wrapper = shallow(<Details />);
        expect(wrapper.find('[data-testid="detailtable"]').length).toBe(1);
        expect(wrapper.find('[data-testid="detailtable"]').text()).toBe("Details");

    });

    //test case 04. render thumbnail image

    test('render details ListDetailImage',() => {
        const wrapper = shallow(<Details />);
        expect(wrapper.find('[data-testid="thumbnailimage"]').length).toBe(1);
        expect(wrapper.find('[data-testid="thumbnailimage"]').text()).toBe("");
      
    });

    //test case 05. render picture
    test('render album picture',() => {
      const wrapper = shallow(<Details />);
      expect(wrapper.find('[data-testid="albumpicture"]').length).toBe(1);
    
     });

     //render without crashing (snapshot)
     test('render without crashing', () => {
      const wrapper = shallow(<Details  />);
      expect(toJson(wrapper)).toMatchSnapshot();
    
  
  });


  
});
