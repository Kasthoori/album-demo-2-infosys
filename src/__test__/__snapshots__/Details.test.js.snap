// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Details Component render without crashing 1`] = `
<div>
  <span
    data-testid="headerdetails"
  >
    <Header />
  </span>
  <br />
  <br />
  <div
    className="container"
  >
    <div
      className="row"
    >
      <div
        className="col-md-4"
        data-testid="detailtable"
      >
        <ForwardRef
          cellspacing="0"
          className="table table-striped table-bordered table-sm"
        >
          <thead>
            <tr>
              <th>
                Details
              </th>
            </tr>
          </thead>
          <tbody>
            <tr>
              <span
                data-testid="thumbnailimage"
              />
            </tr>
          </tbody>
        </ForwardRef>
      </div>
    </div>
    <div
      className="col-md-8"
    >
      <div>
        <h2>
          Album ID: 
        </h2>
        <Image
          alt=""
          data-testid="albumpicture"
          fluid={true}
          rounded={false}
          roundedCircle={false}
          src=""
          thumbnail={false}
        />
        <span
          id="idtext"
        >
          Image ID: 
        </span>
        <br />
        <span>
          Title: 
        </span>
      </div>
    </div>
  </div>
</div>
`;
